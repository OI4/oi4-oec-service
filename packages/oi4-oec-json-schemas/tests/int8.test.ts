import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/dataTypes/int8.schema.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  })
)

describe('int8 schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect(-128).toMatchSchema(schema)
    expect(-127).toMatchSchema(schema)
    expect(-1).toMatchSchema(schema)
    expect(0).toMatchSchema(schema)
    expect(1).toMatchSchema(schema)
    expect(1.0).toMatchSchema(schema)
    expect(126).toMatchSchema(schema)
    expect(127).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    expect(-129).not.toMatchSchema(schema)
    expect(-1.5).not.toMatchSchema(schema)
    expect(128).not.toMatchSchema(schema)

    expect(() => {
      expect(-129).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('-129')
    expect(() => {
      expect(-1.5).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('-1.5')
    expect(() => {
      expect(128).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('128')
  })
})
