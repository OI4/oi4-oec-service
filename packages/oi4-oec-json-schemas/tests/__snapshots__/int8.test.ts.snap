// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`int8 schema test invalid values: -1.5 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be integer
    Path:     dataTypes/int8.schema.json#/type
"
`;

exports[`int8 schema test invalid values: -129 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be >= -128
    Path:     dataTypes/int8.schema.json#/minimum
"
`;

exports[`int8 schema test invalid values: 128 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be <= 127
    Path:     dataTypes/int8.schema.json#/maximum
"
`;
